types/key/node.go

```diff
@@ -265,6 +265,14 @@ func (k NodePublic) Less(other NodePublic) bool {
 	return bytes.Compare(k.k[:], other.k[:]) < 0
 }
 
+// Equal reports whether k is bytewise equal to an other NodePublic.
+//
+// Not that this function can return false for two different encodings
```

Comment:
typo, not->note. But also, I don't understand what this sentence is trying to say. You're comparing one NodePublic to another, there should not be any way to end up with encoding differences, or this package is buggy and needs fixing.